crypto/keyring/keyring.go

```diff
@@ -58,6 +58,9 @@ var (
 type Keyring interface {
 	// Get the backend type used in the keyring config: "file", "os", "kwallet", "pass", "test", "memory".
 	Backend() string
+
+	// Get the db keyring used in the keystore.
+	DB() keyring.Keyring
```

Comment:
IMO we should not modify the existing interface and we should create a new interface that is:

```type KeyringWithDB interface {
     Keyring
     
     // Get the db keyring used in the keystore.
     DB() keyring.Keyring
}


We should also add type assertions that our `keystore` type implements both the `Keyring` and `KeyringWithDB`